### PR TITLE
mongodb-7_0: init at 7.0.12

### PR DESCRIPTION
**Pull request by drupol**
Work in progress:

Trying to use stuff from Arch AUR: https://aur.archlinux.org/cgit/aur.git/tree/?h=mongodb

No luck so far.

build currently broken:

```
Compiling build/nixos/third_party/libmongocrypt/dist/src/mc-efc.o
Compiling build/nixos/mongo/db/commands/fle2_compact_gen.o
In file included from src/mongo/db/catalog/util/partitioned.h:43,
                 from src/mongo/db/query/partitioned_cache.h:32,
                 from src/mongo/db/query/query_stats/query_stats_on_parameter_change.h:34,
                 from build/nixos/mongo/db/query/query_knobs_gen.h:29,
                 from src/mongo/db/pipeline/javascript_execution.h:36,
                 from src/mongo/db/pipeline/expression_context.h:45,
                 from src/mongo/db/pipeline/expression.h:50,
                 from src/mongo/db/pipeline/group_from_first_document_transformation.h:32,
                 from src/mongo/db/pipeline/group_from_first_document_transformation.cpp:30:
src/mongo/util/aligned.h:170:88: warning: use of ‘std::hardware_destructive_interference_size’ [-Winterference-size]
  170 |             aligned_detail::roundUpByStep<stdx::hardware_destructive_interference_size>(sizeof(T))>;
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
src/mongo/util/aligned.h:170:88: note: its value can vary between compiler versions or with different ‘-mtune’ or ‘-mcpu’ flags
src/mongo/util/aligned.h:170:88: note: if this use is part of a public ABI, change it to instead use a constant variable you define
src/mongo/util/aligned.h:170:88: note: the default value for the current CPU tuning is 64 bytes
src/mongo/util/aligned.h:170:88: note: you can stabilize this value with ‘--param hardware_destructive_interference_size=64’, or disable this warning with ‘-Wno-interference-size’
src/mongo/util/aligned.h:170:99: warning: use of ‘std::hardware_destructive_interference_size’ [-Winterference-size]
  170 |             aligned_detail::roundUpByStep<stdx::hardware_destructive_interference_size>(sizeof(T))>;
      |                                                                                                   ^
src/mongo/util/aligned.h:205:85: warning: use of ‘std::hardware_destructive_interference_size’ [-Winterference-size]
  205 |                                        stdx::hardware_constructive_interference_size>;
      |                                                                                     ^
src/mongo/util/aligned.h:205:85: warning: use of ‘std::hardware_destructive_interference_size’ [-Winterference-size]
Compiling build/nixos/mongo/db/geo/geometry_container.o
Compiling build/nixos/mongo/client/connpool.o
In file included from src/third_party/s2/s2.h:19,
                 from src/third_party/s2/s2region.h:6,
                 from src/third_party/s2/s2regionunion.h:12,
                 from src/mongo/db/geo/geometry_container.h:32,
                 from src/mongo/db/geo/geometry_container.cpp:31:
src/third_party/s2/hash.h:8:23: error: invalid use of template-name ‘std::unordered_set’ without an argument list
    8 | #define hash_set std::unordered_set
      |                       ^~~~~~~~~~~~~
In file included from /nix/store/zw4dkm2hl72kfz7j2ci4qbc0avgxzz75-gcc-13.3.0/include/c++/13.3.0/unordered_set:41,
                 from src/third_party/s2/hash.h:7:
/nix/store/zw4dkm2hl72kfz7j2ci4qbc0avgxzz75-gcc-13.3.0/include/c++/13.3.0/bits/unordered_set.h:104:11: note: ‘template<class _Value, class _Hash, class _Pred, class _Alloc> class std::unordered_set’ declared here
  104 |     class unordered_set
      |           ^~~~~~~~~~~~~
src/third_party/s2/hash.h:5:23: error: invalid use of template-name ‘std::unordered_map’ without an argument list
    5 | #define hash_map std::unordered_map
      |                       ^~~~~~~~~~~~~
In file included from /nix/store/zw4dkm2hl72kfz7j2ci4qbc0avgxzz75-gcc-13.3.0/include/c++/13.3.0/unordered_map:41,
                 from src/third_party/s2/hash.h:4:
/nix/store/zw4dkm2hl72kfz7j2ci4qbc0avgxzz75-gcc-13.3.0/include/c++/13.3.0/bits/unordered_map.h:109:11: note: ‘template<class _Key, class _Tp, class _Hash, class _Pred, class _Alloc> class std::unordered_map’ declared here
  109 |     class unordered_map
      |           ^~~~~~~~~~~~~
In file included from src/mongo/crypto/fle_crypto.h:40:
src/mongo/base/data_range.h: In substitution of ‘template<class T> struct mongo::ConstDataRange::ContiguousContainerOfByteLike<T, std::void_t<decltype (declval<T>().data()), typename std::enable_if<(isByteV<typename T::value_type> && mongo::ConstDataRange::HasDataSize>
src/mongo/base/data_range.h:112:97:   required by substitution of ‘template<class Container, typename std::enable_if<mongo::ConstDataRange::ContiguousContainerOfByteLike<Container>::value, int>::type <anonymous> > mongo::ConstDataRange::ConstDataRange(const Container&>
src/mongo/db/query/fle/encrypted_predicate.h:71:76:   required from ‘T mongo::fle::parseFindPayload(BSONValue) [with T = mongo::ParsedFindEqualityPayload; BSONValue = mpark::variant<mongo::BSONElement, std::reference_wrapper<mongo::Value> >]’
src/mongo/db/query/fle/equality_predicate.cpp:49:83:   required from here
src/mongo/base/data_range.h:69:48: note: field ‘const char* mongo::BSONElement::data’ can be accessed via ‘const char* mongo::BSONElement::rawdata() const’
   69 |         std::void_t<decltype(std::declval<T>().data()),
      |                              ~~~~~~~~~~~~~~~~~~^~~~
      |                              rawdata()
scons: *** [build/nixos/mongo/db/geo/geometry_container.o] Error 1
scons: building terminated because of errors.
build/nixos/mongo/db/geo/geometry_container.o failed: Error 1
```

Same error with GCC 11:

```
mongodb> Compiling build/nixos/mongo/util/net/ocsp/ocsp_manager.o
mongodb> In file included from src/third_party/s2/s2.h:19,
mongodb>                  from src/third_party/s2/s2region.h:6,
mongodb>                  from src/third_party/s2/s2regionunion.h:12,
mongodb>                  from src/mongo/db/geo/geometry_container.h:32,
mongodb>                  from src/mongo/db/matcher/expression_geo.h:34,
mongodb>                  from src/mongo/db/matcher/expression_geo.cpp:31:
mongodb> src/third_party/s2/hash.h:8:23: error: invalid use of template-name ‘std::unordered_set’ without an argument list
mongodb>     8 | #define hash_set std::unordered_set
mongodb>       |                       ^~~~~~~~~~~~~
mongodb> In file included from /nix/store/z2hfvpgziara1vfijksjr3adgjl9307y-gcc-11.4.0/include/c++/11.4.0/unordered_set:47,
mongodb>                  from src/third_party/s2/hash.h:7,
mongodb>                  from src/third_party/s2/s2.h:19,
mongodb>                  from src/third_party/s2/s2region.h:6,
mongodb>                  from src/third_party/s2/s2regionunion.h:12,
mongodb>                  from src/mongo/db/geo/geometry_container.h:32,
mongodb>                  from src/mongo/db/matcher/expression_geo.h:34,
mongodb>                  from src/mongo/db/matcher/expression_geo.cpp:31:
mongodb> /nix/store/z2hfvpgziara1vfijksjr3adgjl9307y-gcc-11.4.0/include/c++/11.4.0/bits/unordered_set.h:97:11: note: ‘template<class _Value, class _Hash, class _Pred, class _Alloc> class std::unordered_set’ declared here
mongodb>    97 |     class unordered_set
mongodb>       |           ^~~~~~~~~~~~~
mongodb> In file included from src/third_party/s2/s2.h:19,
mongodb>                  from src/third_party/s2/s2region.h:6,
mongodb>                  from src/third_party/s2/s2regionunion.h:12,
mongodb>                  from src/mongo/db/geo/geometry_container.h:32,
mongodb>                  from src/mongo/db/matcher/expression_geo.h:34,
mongodb>                  from src/mongo/db/matcher/expression_geo.cpp:31:
mongodb> src/third_party/s2/hash.h:5:23: error: invalid use of template-name ‘std::unordered_map’ without an argument list
mongodb>     5 | #define hash_map std::unordered_map
mongodb>       |                       ^~~~~~~~~~~~~
mongodb> In file included from /nix/store/z2hfvpgziara1vfijksjr3adgjl9307y-gcc-11.4.0/include/c++/11.4.0/unordered_map:47,
mongodb>                  from /nix/store/z2hfvpgziara1vfijksjr3adgjl9307y-gcc-11.4.0/include/c++/11.4.0/functional:61,
mongodb>                  from /nix/store/z2hfvpgziara1vfijksjr3adgjl9307y-gcc-11.4.0/include/c++/11.4.0/pstl/glue_algorithm_defs.h:13,
mongodb>                  from /nix/store/z2hfvpgziara1vfijksjr3adgjl9307y-gcc-11.4.0/include/c++/11.4.0/algorithm:74,
mongodb>                  from src/third_party/s2/s2.h:6,
mongodb>                  from src/third_party/s2/s2region.h:6,
mongodb>                  from src/third_party/s2/s2regionunion.h:12,
mongodb>                  from src/mongo/db/geo/geometry_container.h:32,
mongodb>                  from src/mongo/db/matcher/expression_geo.h:34,
mongodb>                  from src/mongo/db/matcher/expression_geo.cpp:31:
mongodb> /nix/store/z2hfvpgziara1vfijksjr3adgjl9307y-gcc-11.4.0/include/c++/11.4.0/bits/unordered_map.h:102:11: note: ‘template<class _Key, class _Tp, class _Hash, class _Pred, class _Alloc> class std::unordered_map’ declared here
mongodb>   102 |     class unordered_map
mongodb>       |           ^~~~~~~~~~~~~
mongodb> Compiling build/nixos/third_party/mozjs/platform/x86_64/linux/build/gc/Unified_cpp_js_src_gc1.o
mongodb> Compiling build/nixos/mongo/util/signal_handlers_synchronous.o
mongodb> In file included from src/third_party/mozjs/extract/js/src/vm/Caches.h:17,
mongodb>                  from src/third_party/mozjs/extract/js/src/vm/Runtime.h:49,
mongodb>                  from src/third_party/mozjs/extract/js/src/gc/PublicIterators.h:19,
mongodb>                  from src/third_party/mozjs/extract/js/src/gc/GCAPI.cpp:17,
mongodb>                  from src/third_party/mozjs/platform/x86_64/linux/build/gc/Unified_cpp_js_src_gc1.cpp:2:
mongodb> src/third_party/mozjs/extract/js/src/frontend/ScopeBindingCache.h:190:32: warning: ‘virtual js::frontend::BindingMap<JSAtom*>* js::frontend::ScopeBindingCache::lookupScope(js::Scope*, js::frontend::ScopeBindingCache::CacheGeneration)’ was hidden [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Woverloaded-virtual-Woverloaded-virtual8;;]
mongodb>   190 |   virtual BindingMap<JSAtom*>* lookupScope(Scope* ptr, CacheGeneration gen);
mongodb>       |                                ^~~~~~~~~~~
mongodb> src/third_party/mozjs/extract/js/src/frontend/ScopeBindingCache.h:238:38: note:   by ‘virtual js::frontend::BindingMap<js::frontend::TaggedParserAtomIndex>* js::frontend::StencilScopeBindingCache::lookupScope(js::frontend::ScopeStencilRef, js::frontend::ScopeBindingCache::CacheGeneration)’
mongodb>   238 |   BindingMap<TaggedParserAtomIndex>* lookupScope(ScopeStencilRef ref,
mongodb>       |                                      ^~~~~~~~~~~
mongodb> src/third_party/mozjs/extract/js/src/frontend/ScopeBindingCache.h:183:32: warning: ‘virtual js::frontend::BindingMap<JSAtom*>* js::frontend::ScopeBindingCache::createCacheFor(js::Scope*)’ was hidden [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Woverloaded-virtual-Woverloaded-virtual8;;]
mongodb>   183 |   virtual BindingMap<JSAtom*>* createCacheFor(Scope* ptr);
mongodb>       |                                ^~~~~~~~~~~~~~
mongodb> src/third_party/mozjs/extract/js/src/frontend/ScopeBindingCache.h:236:38: note:   by ‘virtual js::frontend::BindingMap<js::frontend::TaggedParserAtomIndex>* js::frontend::StencilScopeBindingCache::createCacheFor(js::frontend::ScopeStencilRef)’
mongodb>   236 |   BindingMap<TaggedParserAtomIndex>* createCacheFor(
mongodb>       |                                      ^~~~~~~~~~~~~~
mongodb> src/third_party/mozjs/extract/js/src/frontend/ScopeBindingCache.h:178:16: warning: ‘virtual bool js::frontend::ScopeBindingCache::canCacheFor(js::Scope*)’ was hidden [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Woverloaded-virtual-Woverloaded-virtual8;;]
mongodb>   178 |   virtual bool canCacheFor(Scope* ptr);
mongodb>       |                ^~~~~~~~~~~
mongodb> src/third_party/mozjs/extract/js/src/frontend/ScopeBindingCache.h:235:8: note:   by ‘virtual bool js::frontend::StencilScopeBindingCache::canCacheFor(js::frontend::ScopeStencilRef)’
mongodb>   235 |   bool canCacheFor(ScopeStencilRef ref) override;
mongodb>       |        ^~~~~~~~~~~
mongodb> src/third_party/mozjs/extract/js/src/frontend/ScopeBindingCache.h:191:46: warning: ‘virtual js::frontend::BindingMap<js::frontend::TaggedParserAtomIndex>* js::frontend::ScopeBindingCache::lookupScope(js::frontend::ScopeStencilRef, js::frontend::ScopeBindingCache::CacheGeneration)’ was hidden [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Woverloaded-virtual-Woverloaded-virtual8;;]
mongodb>   191 |   virtual BindingMap<TaggedParserAtomIndex>* lookupScope(ScopeStencilRef ref,
mongodb>       |                                              ^~~~~~~~~~~
mongodb> src/third_party/mozjs/extract/js/src/frontend/ScopeBindingCache.h:266:24: note:   by ‘virtual js::frontend::BindingMap<JSAtom*>* js::frontend::RuntimeScopeBindingCache::lookupScope(js::Scope*, js::frontend::ScopeBindingCache::CacheGeneration)’
mongodb>   266 |   BindingMap<JSAtom*>* lookupScope(Scope* ptr, CacheGeneration gen) override;
mongodb>       |                        ^~~~~~~~~~~
mongodb> src/third_party/mozjs/extract/js/src/frontend/ScopeBindingCache.h:184:46: warning: ‘virtual js::frontend::BindingMap<js::frontend::TaggedParserAtomIndex>* js::frontend::ScopeBindingCache::createCacheFor(js::frontend::ScopeStencilRef)’ was hidden [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Woverloaded-virtual-Woverloaded-virtual8;;]
mongodb>   184 |   virtual BindingMap<TaggedParserAtomIndex>* createCacheFor(
mongodb>       |                                              ^~~~~~~~~~~~~~
mongodb> src/third_party/mozjs/extract/js/src/frontend/ScopeBindingCache.h:265:24: note:   by ‘virtual js::frontend::BindingMap<JSAtom*>* js::frontend::RuntimeScopeBindingCache::createCacheFor(js::Scope*)’
mongodb>   265 |   BindingMap<JSAtom*>* createCacheFor(Scope* ptr) override;
mongodb>       |                        ^~~~~~~~~~~~~~
mongodb> src/third_party/mozjs/extract/js/src/frontend/ScopeBindingCache.h:179:16: warning: ‘virtual bool js::frontend::ScopeBindingCache::canCacheFor(js::frontend::ScopeStencilRef)’ was hidden [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Woverloaded-virtual-Woverloaded-virtual8;;]
mongodb>   179 |   virtual bool canCacheFor(ScopeStencilRef ref);
mongodb>       |                ^~~~~~~~~~~
mongodb> src/third_party/mozjs/extract/js/src/frontend/ScopeBindingCache.h:264:8: note:   by ‘virtual bool js::frontend::RuntimeScopeBindingCache::canCacheFor(js::Scope*)’
mongodb>   264 |   bool canCacheFor(Scope* ptr) override;
mongodb>       |        ^~~~~~~~~~~
mongodb> Compiling build/nixos/mongo/db/log_process_details.o
mongodb> scons: *** [build/nixos/mongo/db/matcher/expression_geo.o] Error 1
mongodb> scons: building terminated because of errors.
mongodb> build/nixos/mongo/db/matcher/expression_geo.o failed: Error 1
```

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by drupol** [comment]
@ofborg build mongodb-7_0

**Comment by drupol** [comment]
@ofborg build mongodb-7_0

**Comment by drupol** [comment]
@ofborg build mongodb-7_0

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -25586,6 +25586,19 @@ with pkgs;
...
+    sasl = cyrus_sasl;
+    boost = boost.override { enableShared = false; };
+    inherit (darwin.apple_sdk.frameworks) CoreFoundation Security;
+    stdenv = if stdenv.isDarwin then
+      darwin.apple_sdk_11_0.stdenv.override (old: {
+        hostPlatform = old.hostPlatform // { darwinMinVersion = "10.14"; };
+        buildPlatform = old.buildPlatform // { darwinMinVersion = "10.14"; };
+        targetPlatform = old.targetPlatform // { darwinMinVersion = "10.14"; };
+      }) else
+      if stdenv.cc.isClang then llvmPackages.stdenv else gcc12Stdenv;
```
Comment (Artturin) [inline pkgs/top-level/all-packages.nix:25599]:
```suggestion
      if stdenv.cc.isClang then llvmPackages.stdenv else gcc14Stdenv;
```

**Comment by drupol** [comment]
@ofborg build mongodb-7_0

**Comment by drupol** [comment]
@ofborg build mongodb-7_0

**Comment by drupol** [comment]
@ofborg build mongodb-7_0

**Comment by drupol** [comment]
@ofborg build mongodb-7_0

**Comment by drupol** [comment]
@ofborg build mongodb-7_0

**Comment by drupol** [comment]
@ofborg build mongodb-7_0

**Comment by drupol** [comment]
@ofborg build mongodb-7_0

**Comment by drupol** [comment]
@ofborg build mongodb-7_0

**Comment by drupol** [comment]
@ofborg build mongodb-7_0

**Comment by drupol** [comment]
@ofborg build mongodb-7_0

**Comment by drupol** [comment]
@ofborg build mongodb-7_0

**Comment by emilazy** [comment]
It’s failing inside the SpiderMonkey (mozjs) build. I bet it would go better if we could use our packaged SpiderMonkey instead. Do you know if the build system has an option for that, like it does for other system libraries?

**Comment by emilazy** [comment]
There is a `use-system-all` option we probably want to set. It’s unclear if they expose it for SpiderMonkey though…

**Comment by emilazy** [comment]
Oh actually, the failure is in whatever `s2` is. Looking into it.

**Comment by drupol** [comment]
Work has been made, closing this.

**Pull request closed without merging**